src/js/pages/WebController/index.js

```diff
@@ -274,26 +273,18 @@ export default class WebControllerPage extends React.PureComponent {
                   pattern="[A-Z,a-z]{3}-[A-Z,a-z]{3}"
                   onKeyUp={e => {
                     const typedValue = e.currentTarget.value;
-                    if (isMobile) {
-                      const parsedValue = typedValue.match('^[A-Z,a-z]{4}');
-                      const isParsedValueExist = parsedValue
-                        ? parsedValue[0] === typedValue
-                        : false;
-                      if (isParsedValueExist) {
-                        const lastChar = typedValue.slice(-1);
-                        e.currentTarget.value =
-                          typedValue.slice(0, 3) + '-' + lastChar;
-                      }
-                    } else {
-                      const typedChar = e.key;
-                      const parsedValue = typedValue.match('^[A-Z,a-z]{3}');
-                      const isParsedValueExist = parsedValue
-                        ? parsedValue[0] === typedValue
-                        : false;
-                      if (isParsedValueExist && typedChar !== 'Backspace') {
-                        e.currentTarget.value = typedValue + '-';
-                      }
+                    const typedChar = e.key;
+                    const hasHyphen = typedValue.includes('-');
+                    const parsedValue = typedValue.match('^[A-Z,a-z]{3}');
+                    const isParsedValueExist = parsedValue
+                      ? parsedValue[0] === typedValue
+                      : false;
+                    if (isParsedValueExist && typedChar !== 'Backspace') {
+                      e.currentTarget.value = typedValue + '-';
+                    } else if (typedChar === '-' && hasHyphen) {
+                      e.currentTarget.value = typedValue.replace(/-+/g, '-');
```

Comment:
@yadvirkaur I might be wrong here but we are replacing `/-+/g` both `-` and `+` aint? Or does it represents `-` n number of times? if so, can you please update implementation a bit to be more clear. It can cause confusion.

Reply:
The expression `(/-+/g, '-')` here matches one or more hyphens and replaces them with a single hyphen. It's not related to the `+` character itself. To make it clearer, I can update it to` (/[-]+/g, '-')`. Let me know if that works!

Reply:
@yadvirkaur yes that works perfectly fine